Pod/Classes/URBNDataSourceAdapter.m

```diff
@@ -32,26 +32,201 @@
 NSString *const URBNSupplementaryViewKindHeader = @"URBNSupplementaryViewKindHeader";
 NSString *const URBNSupplementaryViewKindFooter = @"URBNSupplementaryViewKindFooter";
 
+/**
+ *  These are our internal objects to wrap up the collectionView dataSource delegate stuff.
+ */
+@interface URBNDataSourceTableViewDataSourceDelegate: NSObject <UITableViewDataSource, UITableViewDelegate>
+@property (nonatomic, weak) URBNDataSourceAdapter *ds;
+@end
+@interface URBNDataSourceCollectioniewDataSourceDelegate: NSObject <UICollectionViewDataSource, UICollectionViewDelegateFlowLayout>
+@property (nonatomic, weak) URBNDataSourceAdapter *ds;
+@end
+
 @interface URBNDataSourceAdapter ()
 
 @property (nonatomic, strong) NSMutableDictionary *cellConfigurationBlocks;
 @property (nonatomic, strong) NSMutableDictionary *viewConfigurationBlocks;
 
+@property (nonatomic, strong) NSMutableArray *dataSources;
+@property (nonatomic, strong) NSMutableDictionary *prototypeCells;
+@property (nonatomic, strong) NSMutableDictionary *prototypeHeaders;
+
+- (CGSize)heightForRowAtIndexPath:(NSIndexPath *)indexPath;
+- (CGSize)heightForSupplementaryViewOfType:(URBNSupplementaryViewType)type atIndexPath:(NSIndexPath *)indexPath;
+
 @end
 
 @implementation URBNDataSourceAdapter
 
 #pragma mark - init
 - (instancetype)init {
-    if ((self = [super init])) {
+    self = [super init];
+    if (self) {
         self.rowAnimation = UITableViewRowAnimationAutomatic;
         self.cellConfigurationBlocks = [NSMutableDictionary dictionary];
         self.viewConfigurationBlocks = [NSMutableDictionary dictionary];
+        self.dataSources = [NSMutableArray array];
+        self.prototypeCells = [NSMutableDictionary dictionary];
+        self.prototypeHeaders = [NSMutableDictionary dictionary];
+        URBNDataSourceCollectioniewDataSourceDelegate *cd = [URBNDataSourceCollectioniewDataSourceDelegate new];
+        URBNDataSourceTableViewDataSourceDelegate *td = [URBNDataSourceTableViewDataSourceDelegate new];
+        cd.ds = self;
+        td.ds = self;
+        [self.dataSources addObject:cd];
+        [self.dataSources addObject:td];
     }
-    
     return self;
 }
 
+#pragma mark - Setters
+- (void)setFallbackDataSource:(id)fallbackDataSource {
+    if (_fallbackDataSource == fallbackDataSource) {
+        return;
+    }
+    _fallbackDataSource = fallbackDataSource;
+    [self.dataSources addObject:fallbackDataSource];
```

Comment:
So I know this _shouldn't_ happen but what happens when someone sets a fallbackdatasource and then resets it? And what happens when they set it to `nil`